src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js

```diff
@@ -73,6 +74,33 @@ angular.module('esn.inbox.libs')
       return mailbox;
     }
 
+    function mailboxtoTree(arr) {
+      const arrMap = new Map(arr.map(item => [item.id, item]));
+      const tree = [];
+
+      for (let i = 0, l = arr.length; i < l; i++) {
+        const item = arr[i];
+
+        if (item.parentId) {
+          const parentItem = arrMap.get(item.parentId);
+
+          if (parentItem) {
+            const { nodes } = parentItem;
+
+            if (nodes) {
+              parentItem.nodes.push(item);
+            } else {
+              parentItem.nodes = [item];
+            }
```

Comment:
Does your algorithm preserve ordering of nodes?

Reply:
it's already done before, in an other function in the service

Reply:
Yes but your algorithm needs to preserve this property. Have you tested it in regard of sorting?